### DISCUSSION TITLE
Will cinnamon ever support wayland? x11 is going to be discontinued in the future and will just get more broken.

### DISCUSSION THREAD

**Discussion by dnkmmr69420** [linuxmint/discussions#24] [Questions] [answered]
Like cinnamon should start migrating to wayland

**Comment by arxo** [comment]
The issue of Wayland support has been raised multiple times over the years, e.g.
https://github.com/linuxmint/cinnamon/issues/10787
https://github.com/linuxmint/cinnamon/issues/10136
https://github.com/linuxmint/cinnamon/issues/10078
https://github.com/linuxmint/cinnamon/issues/10025
https://github.com/linuxmint/cinnamon/issues/8512
https://github.com/linuxmint/cinnamon/issues/5201 .
All of those tickets were closed without any actual work being done in terms of implementing Wayland support.

However, with several other popular desktop environments (e.g. KDE and Gnome) already supporting Wayland
or development at least being underway (e.g. XFCE) due to the winding down of X11 development and sufficient maturity of Wayland,
I think it is time to acknowledge the elephant in the spice store and start tracking Wayland support or lack thereof in Cinnamon.

For starters, it would be good to have a list of issues that need to be addressed to make this work.

**Reply by LinuxOnTheDesktop** [reply]
I believe that Mint's mutter-rebase was motivated partly to make a future transition to Wayland possible. Also, Clem has written that Mint will support Wayland - but that claim does not obviate your question of what work, aside from the rebase, will need to be done and when it will be done.

**Comment by axel358** [comment]
It's not the fact of whether Wayland is mature enough or not. Is the fact that x11 is in vegetative state rn. And pulling the plug it's already on the table. Gtk5 might drop the x11 backend n both GNOME n Plasma are working on getting rid of the xorg dependency. RHEL is already deprecating it n Fedora might soon follow. Even MATE n xfce are discussing Wayland support. So I  agree that might be time to stop dodging the bullet n sit n talk about this. KDE has shown that Wayland doesn't mean the death of the traditional desktop metaphor

**Comment by ghost** [comment]
Waydroid and all other apps work only on wayland, Anbox is now depreceated, does not work anymore, cinnamon shall support wayland as if one wants android apps support, then he'd have to install another DE, except two apps, every app supports wayland in MATE, i know migration to wayland is not easy, especially if you see the NVIDIA drivers, they simply don't play nice with wayland, xapps don't support wayland, literally xorg is just a patchwork, it was good at its time.

**Reply by Reno-Sifana** [reply]
But some programs such as Redshift doesn't support Wayland or Mir yet. [Redshift Github's Project Page](https://github.com/jonls/redshift/). Linux Mint's relying Redshift to adjust color temperature at the night. Is good idea if Linux Mint's Developers will add the capabilities into cinnamon-settings-daemon for example in Linux Mint 22 next year.
(Notes: GNOME And KDE Plasma 5 already have night color settings feature long time ago. As Far As I Know)

**Comment by clefebvre** [comment, marked as answer]
Pull requests are welcome.

**Reply by LinuxOnTheDesktop** [reply]
That makes it sound as if Mint is waiting for people outside of Mint to do the relevant work.

**Reply by anaximeno** [reply]
> That makes it sound as if Mint is waiting for people outside of Mint to do the relevant work.

Well, Mint is a community-driven distribution, so it's natural for people to help with the transition.

**Reply by axel358** [reply]
There is no 'outside of Mint' we're all Mint, both users and devs, if we make demands as big as this one we need to be willing to help achieve it

**Reply by arxo** [reply]
@clefebvre At the very least we need a list of issues that need to be addressed to make Wayland happen for Linux Mint.

On a closely related note the Solus project recently announced deprecation of MATE support on their distro with these words:
> After evaluating MATE, we have concluded that it does not have a credible and active Wayland strategy, with the project itself effectively being on life support. As such, we have decided to sunset the MATE Edition.
(https://getsol.us/2023/07/08/solus-4-4-released/)

Looks like the same can be said about Cinnamon and Linux Mint in general (not the "on life support" bit, but "pull requests welcome" doesn't really strike me as a "credible and active" Wayland strategy).
